### ISSUE TITLE
Calling flake8-type-checking explicitly

### ISSUE THREAD

**Issue by danielcohenlive** [snok/flake8-type-checking#148]
I'm interested in running this in an environment that uses buck, not pip, and I'm not sure I have the ability to modify the way flake8 works for the whole repo.  Linting is done by a different package than the one I own.  Is there any way I could set up a python test case or something that I could explicitly call flake8-type-checking from?  What would that look like if so?

Thanks!

**Comment by sondrelg** [comment]
Yes I think that would be pretty simple to get working. Essentially flake8 just passes the parsed python code from a file to the plugin, not much else.

Something like this should work:

```python
import ast
from flake8_type_checking.checker import TypingOnlyImportsChecker


# you could replace this with loading the code from the file directly, etc.
# this is just an example of code loaded as a string
my_code = """  
from pandas import DataFrame

foo: DataFrame
"""

# Parse the python code as an [AST](https://docs.python.org/3/library/ast.html)
tree = ast.parse(my_code)

# Load the checker from the plugin
options = {'type_checking_strict': False}  # if you want to specify options, add them here
visitor = TypingOnlyImportsChecker(tree, options)

for e in visitor.errors:
    print(e)
```

So flake8's role in this setup is really just to traverse the files that are being linted and feeding them to the checker. The checker is otherwise possible to run independently :+1: 

Was that clear? Let me know if you have any other questions :slightly_smiling_face:

**Comment by danielcohenlive** [comment]
Yes, that's clear.  Thanks!

**Comment by danielcohenlive** [comment]
Is there any way to mark lines to not be evaluated?

**Comment by danielcohenlive** [comment]
And also, if I want to ignore error types, like "TC200", how would I do that?

**Comment by sondrelg** [comment]
Both error code ignores and line ignores are actually handled by flake8, so there's no easy way to leverage the plugin for this. 

> How is it deciding between things that should be string literals and ones that should be done with from __future__ import annotations

Yeah this is a good point. You need to add `'extend-select': ['TC', 'TC2']` to the option dict I believe

In summary it seems like this would be a lot simpler if you used flake8. Have you tried running flake8 with pre-commit? Is that also not an option?

**Comment by danielcohenlive** [comment]
The issue is just that it's a giant monorepo and my team doesn't own the util running flake8.  But I think I can work around it.

**Comment by sondrelg** [comment]
Good luck :slightly_smiling_face: Feel free to comment here if you have any other questions :+1:
